PG.Commons/PG.Commons/Util/StringUtility.cs

```diff
@@ -96,5 +111,22 @@ private static List<string> SplitStringInternal(string s, char separator, bool s
             list.AddRange(from str in split where !IsNullEmptyOrWhiteSpace(str) select shouldClean ? str.Trim() : str);
             return list;
         }
+
+        /// <summary>
+        /// Removes any file extension form a file or filepath.
+        /// </summary>
+        /// <param name="fileName"></param>
+        /// <returns></returns>
+        /// <exception cref="ArgumentException"></exception>
+        public static string StripFileExtension(string fileName)
+        {
+            if (!StringUtility.HasText(fileName))
```

Comment:
is this necessary? `Path.ChangeExtension` already does the necessary checks? Do you want to have a "local" stack trace?

Reply:
> On Windows-based desktop platforms, if path is `null` or an empty string (`""`), the path information is returned unmodified. 

As far as I understand this part, it does nothing. In our case we throw an exception if that happens.